res/controllers/Vestax-VCI-100MKII-scripts.js

```diff
@@ -158,147 +163,123 @@ VCI102.pitch = function(ch, midino, value, status, group) {
     } else {
-        engine.setValue(group, "reloop_exit", 1);
+        VCI102.meta(ch, value, VCI102.slot(group, 3), "meta");
     }
 };
 
-VCI102.loop_halve = function(ch, midino, value, status, group) {
-    if (engine.getValue(group, "loop_enabled")) {
-        engine.setValue(group, "loop_halve", value > 0);
-    } else if (value) {
-        VCI102.setLoopLength(ch, status, VCI102.loopLength[ch] / 2);
+VCI102.parameter4 = function(ch, midino, value, status, group) {
+    var n = engine.getValue(group, "focused_effect");
+    if (n) {
+        VCI102.parameter(ch, value, VCI102.slot(group, n), "parameter4");
+    } else {
+        VCI102.meta(ch, value, group, "super1");
     }
 };
 
-VCI102.loop_double = function(ch, midino, value, status, group) {
-    if (engine.getValue(group, "loop_enabled")) {
-        engine.setValue(group, "loop_double", value > 0);
-    } else if (value) {
-        VCI102.setLoopLength(ch, status, VCI102.loopLength[ch] * 2);
+VCI102.prev_effect = function(ch, midino, value, status, group) {
+    var key, n = engine.getValue(group, "focused_effect");
+    if (n) {
+        group = VCI102.slot(group, n);
+        key = "prev_effect";
+    } else {
+        key = "prev_chain";
     }
+    if (VCI102.shift[1 - ch % 2]) {
+        // clear effect if shift of the other Deck
+        key = "clear";
+    }
+    engine.setValue(group, key, value > 0);
 };
 
-VCI102.move = function(ch, group, dir) {
-    if (engine.getValue(group, "loop_enabled")) {
-        // move the loop by the current length
-        engine.setValue(
-            group, "loop_move", dir * (
-                engine.getValue(group, "loop_end_position")
-                    - engine.getValue(group, "loop_start_position"))
-                / engine.getValue(group, "track_samplerate")
-                * engine.getValue(group, "file_bpm") / 120);
+VCI102.next_effect = function(ch, midino, value, status, group) {
+    var key, n = engine.getValue(group, "focused_effect");
+    if (VCI102.shift[1 - ch % 2]) {
+        // change focus if shift of the other Deck
+        if (value) {
+            engine.setValue(group, "focused_effect", (n + 1) % 4);
+        }
     } else {
-        // jump by the default length
-        engine.setValue(group, "beatjump", dir * VCI102.loopLength[ch]);
+        if (n) {
+            group = VCI102.slot(group, n);
+            key = "next_effect";
+        } else {
+            key = "next_chain";
+        }
+        engine.setValue(group, key, value > 0);
     }
 };
 
-VCI102.move_backward = function(ch, midino, value, status, group) {
+VCI102.loopSize = [4, 4, 4, 4];  // saved beatloop_size
+
+VCI102.loop = function(ch, midino, value, status, group) {
     if (value) {
-        VCI102.move(ch, group, -1);
+        if (engine.getValue(group, "loop_enabled")) {
+            engine.setValue(group, "reloop_toggle", 1);
+            // restore beatloop_size
+            engine.setValue(group, "beatloop_size", VCI102.loopSize[ch]);
```

Comment:
loop_scale is not intended for use with beatloops. It only takes into consideration the size of the loop, so it will create off beat loops with non constant beatgrids.